From: Sven Eckelmann <sven@narfation.org>
Date: Sun, 14 Oct 2018 20:03:17 +0200
Subject: mac80211: Enable ath9k dynack by default

Signed-off-by: Sven Eckelmann <sven@narfation.org>

Forwarded: no

diff --git a/package/kernel/mac80211/ath.mk b/package/kernel/mac80211/ath.mk
index 64aac41b4d2ce777f587967ae00a6946ff5bc820..8dde01a6504b83475527c7b9b5b2c2972e4d382d 100644
--- a/package/kernel/mac80211/ath.mk
+++ b/package/kernel/mac80211/ath.mk
@@ -104,6 +104,7 @@ define KernelPackage/ath/config
 	config PACKAGE_ATH_DYNACK
 		bool "Enable Dynack support"
 		depends on PACKAGE_kmod-ath9k-common
+		default y
 		help
 		  Enables support for Dynamic ACK estimation, which allows the fastest possible speed
 		  at any distance automatically by increasing/decreasing the max frame ACK time for
diff --git a/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh b/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
index 36aebbb2ccfec2137d5d260fe2111d77f531ddec..a6643948b19753bb10641ae43caed4c6ecac43d4 100644
--- a/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
+++ b/package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh
@@ -761,7 +761,6 @@ drv_mac80211_setup() {
 
 	set_default rxantenna 0xffffffff
 	set_default txantenna 0xffffffff
-	set_default distance 0
 	set_default antenna_gain 0
 
 	[ "$txantenna" = "all" ] && txantenna=0xffffffff
@@ -769,7 +768,12 @@ drv_mac80211_setup() {
 
 	iw phy "$phy" set antenna $txantenna $rxantenna >/dev/null 2>&1
 	iw phy "$phy" set antenna_gain $antenna_gain
-	iw phy "$phy" set distance "$distance"
+
+	if [ -n "$distance" ]; then
+		iw phy "$phy" set distance "$distance"
+	else
+		iw phy "$phy" set distance auto || iw phy "$phy" set distance 0
+	fi
 
 	[ -n "$frag" ] && iw phy "$phy" set frag "${frag%%.*}"
 	[ -n "$rts" ] && iw phy "$phy" set rts "${rts%%.*}"
